README_INTERFACE_TESTS.md

```diff
@@ -0,0 +1,173 @@
+1) on `ingress-relation-joined`, the requirer should write its part of the data to the relation. 
+2) on `ingress-relation-changed`, if the provider has written its part of the data to the relation, then the provider should publish its part of the data; otherwise it should publish no data.
+
+## A minimal test case
+
+Specify the schema for the `ingress` interface databags in `./interfaces/ingress/v0/schema.py`.
+
+Create a python file at `<repo-root>/interfaces/ingress/v0/interface_tests/my_tests.py` with these contents:
+
+```python
+from scenario import State
+from interface_tester.interface_test import interface_test_case
+
+
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='requirer',
+)
+def test_data_published_on_joined(output_state: State):
+    return
+```
+
+This is a complete interface test.
+
+## Validating the output state
+
+If the specification dictates something more than "the databag contents should match this schema in that event", for example:
+
+> on `ingress-relation-joined`, the requirer should write its part of the data to the relation and set its status to Active.
+
+The test would then become:
+
+```python
+from scenario import State
+from interface_tester.interface_test import interface_test_case
+
+
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='requirer',
+)
+def test_data_published_on_joined(output_state: State):
+    assert output_state.status.unit.name == 'active'
+```
+
+
+## Testing a negative
+
+The second norm we have above contains a negative. We split it into two assertions to write the tests:
+
+> on `ingress-relation-changed`, if the requirer has provided 'data' and 'baz' information in its relation databags, then the provider should publish its part of the data.
+> on `ingress-relation-changed`, if the requirer has NOT provided 'data' and 'baz' information in its relation databags, then the provider should publish no data.
+
+This becomes two test cases:
+
+```python
+from scenario import State, Relation
+from interface_tester.interface_test import interface_test_case, SchemaConfig
+
+
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='provider',
+    input_state=State(
+        relations=[Relation(
+            endpoint='foo',
+            interface='ingress',
+            remote_app_name='remote',  # this is our simulated requirer
+            remote_app_data={
+                'data': 'foo-bar',
+                'baz': 'qux'
+            }
+        )]
+    )
+)
+def test_data_published_on_joined_if_remote_has_sent_valid_data(output_state: State):
+    pass
+
+    
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='provider',
+    schema=SchemaConfig.empty,
+    input_state=State(
+        relations=[Relation(
+            endpoint='foo',
+            interface='ingress',
+            remote_app_name='remote',
+            remote_app_data={
+                'some': 'rubbish'
+            }
+        )]
+    )
+
```

Comment:
```suggestion
```

Reply:
This comment was resolved but the fix wasn't applied

(in general, please resolve comments after the fix was made or both parties agreed on it not being fixed)